tgui/packages/tgui/interfaces/TacticalMap.tsx

```diff
@@ -1,24 +1,277 @@
+  'home': () => HomePanel,
+};
+
+const themes = [
+  {
+    'theme': 'default',
+    'button-color': 'black',
+  },
+  {
+    'theme': 'crtblue',
+    'button-color': 'blue',
+  },
+  {
+    'theme': 'xeno',
+    'button-color': 'purple',
+  },
+];
+
+const colorOptions = [
+  'black',
+  'red',
+  'orange',
+  'blue',
+  'purple',
+  'green',
+  'brown',
+];
+
+const colors: Record<string, string> = {
+  'black': '#000000',
+  'red': '#FC0000',
+  'orange': '#F59A07',
+  'blue': '#0561F5',
+  'purple': '#C002FA',
+  'green': '#02c245',
+  'brown': '#5C351E',
+};
+
 export const TacticalMap = (props, context) => {
   const { data, act } = useBackend<TacMapProps>(context);
+
+  const PageComponent = PAGES[data.currentMenu]();
+
   return (
-    <Window title={'Tactical Map'} theme="usmc" width={650} height={680}>
+    <Window
+      width={650}
+      height={750}
+      theme={themes[data.themeId]['theme']}
+      title={'Tactical Map'}>
       <Window.Content>
-        <ByondUi
-          params={{
-            id: data.mapRef,
-            type: 'map',
-          }}
-          class="TacticalMap"
-        />
+        <PageComponent />
       </Window.Content>
     </Window>
   );
 };
+
+const HomePanel = (props, context) => {
+  const { data, act } = useBackend<TacMapProps>(context);
+
+  return (
+    <Section fontSize="20px" textAlign="center" title="Tactical Map Options">
+      <Stack justify="center" align="center" fontSize="15px">
+        <Stack.Item grow>
+          <Button
+            color={themes[data.themeId]['button-color']}
+            content={'tacmap'}
+            icon={'map'}
+            onClick={() =>
+              act('menuSelect', {
+                selection: 'view',
+              })
+            }
+          />
+        </Stack.Item>
+        <Stack.Item grow>
+          <Button
+            color={themes[data.themeId]['button-color']}
+            content={'old canvas'}
+            icon={'eye'}
+            onClick={() =>
+              act('menuSelect', {
+                selection: 'old_canvas',
+              })
+            }
+          />
+        </Stack.Item>
+        <Stack.Item grow>
+          <Button
+            color={themes[data.themeId]['button-color']}
+            content={'new canvas'}
+            icon={'paintbrush'}
+            onClick={() =>
+              act('menuSelect', {
+                selection: 'draw',
+              })
+            }
+          />
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+};
+
+const ViewMapPanel = (props, context) => {
+  const { data, act } = useBackend<TacMapProps>(context);
+  return (
+    <ByondUi
+      params={{
+        id: data.mapRef,
+        type: 'map',
+      }}
+      class="TacticalMap"
+    />
+  );
+};
+
+const OldMapPanel = (props, context) => {
+  const { data, act } = useBackend<TacMapProps>(context);
+  return (
+    data.flatImage && (
+      <DrawnMap svgData={data.svgData} flatImage={data.flatImage} />
+    )
+  );
```

Comment:
> So for xenos it seems data.flatImage is not truthy. For example if I change the return to this:
> 
> ```js
>   return <DrawnMap svgData={data.svgData} flatImage={data.flatImage} />;
> ```
> 
> Then at least xenos do see the drawing and the html isn't empty here: ![image](https://user-images.githubusercontent.com/76988376/270156232-6f8a0651-15fc-4889-94ef-6d494b063ffd.png)
> 
> The code here doesn't need to change persay, but just pointing out thats why xenos always have an empty panel atm because it resolves to false and thus puts nothing.

 - that was by design, I didn't want to overlay a drawing ontop of a non-existent png asset. So if a new user joins and tries to open the tacmap, they wouldn't have access to old generated distributed png asset (icon2html) but they would if they were already in the , I was asking in discord if there was a way to check if an asset is stored on the users end (in the case of new users, it wouldn't be) and then distribute it to them, but I didn't get a response.

Reply:
> The code here doesn't need to change persay, but just pointing out thats why xenos always have an empty panel atm because it resolves to false and thus puts nothing.